File: 프로그래머스/lv1/1845. 폰켓몬/폰켓몬.py
def solution(nums):
    answer = 0
    dict1 = {}
    for num in nums:
        if num not in dict1:
            dict1[num] = 1
    
    answer = len(dict1)
    
    if len(dict1) > len(nums)/2 :
        answer = len(nums)/2
    
    return answer

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.java
import java.util.*;

class Solution {
    private Set<Integer> set = new HashSet<>();
    public int solution(int[] nums) {
        int answer = 0;
        for(int num: nums){
            set.add(num);
        }
        if(set.size() > nums.length / 2){
            return nums.length / 2;
        }
        return set.size();
    }
}